src/developer/web-api/overview.md

```diff
@@ -86,13 +90,226 @@ using SSL/TLS (HTTPS) to encrypt communication with clients. Consider this
 a hard requirement in order to provide secure interactions with the Web
 API.
 
-### Two-factor authentication { #webapi_2fa } 
+## Two-factor authentication { #webapi_2fa } 
 
 DHIS2 supports two-factor authentication. This can be enabled per user.
 When enabled, users will be asked to enter a 2FA code when logging in. You
 can read more about 2FA [here](https://www.google.com/landing/2step/).
 
-### OAuth2 { #webapi_oauth2 } 
+## Personal Access Token { #webapi_pat_authentication }
+Personal access tokens (PATs) are an alternative to using passwords for
+authentication to DHIS2 when using the API.
+
+PATs can be a more secure alternative to HTTP Basic Authentication,
+and should be your preferred choice when creating a new app/script etc. 
+
+HTTP Basic Authentication is considered insecure because, among other things, 
+it sends your username and password in clear text. It may be deprecated in 
```

Comment:
This is not really true though, since the username and password should be encrypted over HTTPS?

Reply:
Yes, that is true. It does not send it in clear text when using HTTPS, but it will if you are using plain HTTP.
The wording should rather be: “it USES your username and password in cleat text”
I will change it to that.

Reply:
I think we just need to be careful here, since use of a token over HTTP is no more secure than a username and password.  The difference of course being that you can revoke the token, but you can also change your password. 

Use of any authentication method over HTTP is not secure. So, I am not sure that there is a big difference between Basic Authentication and PAT over HTTP...the real problem is the lack of use of HTTPS (which of course should never be the case).

Reply:
Well, I think that using a PAT token has mainly the benefit that you don't expose your username and password, which can be used to log in normally. The PATs can only be used via the API, and you can limit the access with the constraints. I agree it's not a lot more secure, but I personally think it is the better alternative if we have to choose.
There are more features we want to add to the PATs in the future as, for example, monitoring and a log of the latest uses etc. More constraints in the authorities, as well as a URL pattern constraint.
I think stressing the use of HTTP and making it maybe "harder" to use the server without, is maybe even more important.
We also intend to introduce other types of token soon like JWT for front-end. 
I also think, making another type of more secure authentication alternative "token" for server to server communication should be implemented.

Reply:
Would agree with all of that. We are in the process of revamping the implementation docs, and a more thorough discussion of configuring HTTPS defintely needs to be there.

I just want to be sure that we are not give a false impression that Basic Authentication is insecure. It most certainly is when used over HTTP, same with a PAT, although the potential damaged with a PAT could be more limited. Regardless, neither should be used over HTTP.  

Completely agree that PATs are the way to go and it sounds like a great feature, but there are still instances in which they are going to have to be used, and just want to be sure we are not giving people a false impression that Basic Authentication is somehow intrinsically insecure, at least over HTTPS when configured correctly.